### PR TITLE
Token holder graph

### PR DESCRIPTION
**Pull request by SidharthK2**
# Add HiIQ Token holder graph

HiIQ token holders over time graph.

## Linked issues

closes #1656, closes #1656, closes https://github.com/EveripediaNetwork/issues/issues/1656

### CODE REVIEW

File: src/app/dashboard/page.tsx [hunk 1]
```diff
@@ -77,6 +96,7 @@ const Home: NextPage = () => {
   const isFetchedData = useRef(false)
   const { tvl } = useErc20()
   const { totalHiiqSupply } = useLockOverview()
+  const [numberOfIQHolder, setNumberOfIQHolder] = useState<number>(0)
```
Comment (Adebesin-Cell) [inline src/app/dashboard/page.tsx:99]: You can remove the number type, TS will infer it.

File: src/app/dashboard/page.tsx [hunk 2]
```diff
@@ -337,9 +337,31 @@ const Home: NextPage = () => {
...
+        <GridItem colSpan={{ base: 12, lg: 12 }}>
+          <Box mb={6}>
+            <GraphComponent
+              getRootProps={getHolderRootProps}
+              areaGraph={false}
+              graphData={transformHiIQHolderData(holderData)}
+              graphCurrentValue={numberOfIQHolder}
+              graphTitle="IQ Token Holders over time"
+              height={120}
+              isHolderGraph={true}
```
Comment (Adebesin-Cell) [inline src/app/dashboard/page.tsx:349]: You don't have to pass ``true`` to ``isHolderGraph``

Comment (SidharthK2) [reply src/app/dashboard/page.tsx:349]: thanks, will change.

File: src/utils/getTokenHoldersCount.ts [hunk 3]
```diff
@@ -1,7 +1,16 @@
...
-    'https://api.ethplorer.io/getTokenInfo/0x579CEa1889991f68aCc35Ff5c3dd0621fF29b0C9?apiKey=freekey',
-  )
-  const data = await res.json()
-  return data.holdersCount
+  try {
+    const res = await fetch(
+      'https://api.ethplorer.io/getTokenInfo/0x579CEa1889991f68aCc35Ff5c3dd0621fF29b0C9?apiKey=freekey',
+    )
+    const data = await res.json()
+    if (!data || typeof data.holdersCount !== 'number') {
```
Comment (Adebesin-Cell) [inline src/utils/getTokenHoldersCount.ts:7]: If there's no data case is better, don't test for the type.

Comment (SidharthK2) [reply src/utils/getTokenHoldersCount.ts:7]: Any specific reason for that?

Comment (Adebesin-Cell) [reply src/utils/getTokenHoldersCount.ts:7]: Type might not come as a number, might be aa ``BigInt`` or ``string``, so it's not safe.

**Pull request merged**